### ISSUE TITLE
pixelPositionForScreenPosition intermittently throws

### ISSUE THREAD

**Issue by chmac** [atom/atom#15689]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

`pixelPositionForScreenPosition()` intermittently throws. Full error reported [here](forgecrafted/rulerz#25), excerpt is:

```
TypeError: Cannot read property 'NaN' of undefined
    at TextEditorComponent.renderedScreenLineForRow (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-component.js:905:31)
    at TextEditorComponent.pixelLeftForRowAndColumn (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-component.js:2229:29)
    at TextEditorComponent.pixelPositionForScreenPosition (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-component.js:187:40)
    at HTMLElement.pixelPositionForScreenPosition (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-element.js:243:32)
    at HTMLElement.RulerView.update (/packages/rulerz/lib/ruler-view.coffee:61:23)
```

This error is intermittently thrown. I've wrapped it in a `try` / `catch` and if I retry a second time inside a `setTimeout(fn)` then I've never been able to get it to throw twice in my testing.

### Steps to Reproduce

* Install rulerz plugin 0.4.3
* Repeatedly open new split screens

Apologies, I know that's not a very good reproduction, this is my first foray into atom hacking so I'm not sure how to produce a smaller reproduction yet.

**Reproduces how often:** [What percentage of the time does it reproduce?]

It always happens on application start, but then intermittently on new pane creation.

### Versions

```
Atom    : 1.19.7
Electron: 1.6.9
Chrome  : 56.0.2924.87
Node    : 7.4.0
```

```
apm  1.18.4
npm  3.10.10
node 6.9.5 x64
python 2.7.10
git 2.10.2
```

### Additional Information

I realise this is a low quality bug report, apologies. I'm open to any feedback on how I might be able to improve it.

**Comment by rsese** [comment]
Thanks for the report!

> This error is intermittently thrown. I've wrapped it in a try / catch and if I retry a second time inside a setTimeout(fn) then I've never been able to get it to throw twice in my testing.

Would you be able to put together a minimal code reproduction for the error?

> Repeatedly open new split screens

About how many splits on doesn't it take before you see the error?  

> It always happens on application start, but then intermittently on new pane creation.

Hmmm, just to clarify, do you mean that with rulerz installed, on application start you always get this error?

About the bug report itself, thanks so much for using the template, that's always a great first step :+1:  A general suggestion would be to share more details and be explicit as possible - e.g. for the steps to reproduce, sharing the exact menu item/keyboard shortcut/etc. for a command is helpful.  Adding screenshots or GIF in this case can be helpful as well and provide more context.

**Comment by chmac** [comment]
@rsese Yes, with `rulerz` version `0.4.3` installed, I see this error on startup.

In terms of a smaller reproduction, I'll try to invest some time in that, but I'd guess it's unlikely as my `try` / `catch` hack is working for now, so my immediate pain point has been fixed. Laziness will most likely prevail. :-)

**Comment by rsese** [comment]
Sounds good @chmac :+1:  I did try to reproduce by installing `0.4.3` of rulerz but didn't see the error on startup - we'll go ahead and close for now but we can re-open later if necessary.

**Comment by chmac** [comment]
@rsese OK, it's potentially a conflict between multiple packages then, much harder to pin down. Will reopen if I ever get a working reproduction.
